server/protocols/jmap/src/main/java/org/apache/james/jmap/model/mailbox/MailboxRequest.java

```diff
@@ -0,0 +1,150 @@
+
+        private Optional<String> id;
+        private String name;
+        private String parentId;
+        private Optional<Role> role;
+        private Optional<SortOrder> sortOrder;
+
+        private Builder() {
+            id = Optional.empty();
+            role = Optional.empty();
+            sortOrder = Optional.empty();
+        }
+
+        public Builder id(String id) {
+            Preconditions.checkNotNull(id);
+            this.id = Optional.of(id);
+            return this;
+        }
+
+        public Builder name(String name) {
+            Preconditions.checkNotNull(name);
+            this.name = name;
+            return this;
+        }
+
+        public Builder parentId(String parentId) {
+            this.parentId = parentId;
+            return this;
+        }
+
+        public Builder role(Optional<Role> role) {
+            this.role = role;
+            return this;
+        }
+
+        public Builder sortOrder(SortOrder sortOrder) {
+            this.sortOrder = Optional.of(sortOrder);
+            return this;
+        }
+
+
+        public MailboxRequest build() {
+            Preconditions.checkState(!Strings.isNullOrEmpty(name), "'name' is mandatory");
+            return new MailboxRequest(id, name, Optional.ofNullable(parentId), role, sortOrder);
+        }
+    }
+
+    private final Optional<String> id;
+    private final String name;
+    private final Optional<String> parentId;
+    private final Optional<Role> role;
+    private final Optional<SortOrder> sortOrder;
+
+    @VisibleForTesting
+    MailboxRequest(Optional<String> id, String name, Optional<String> parentId, Optional<Role> role, Optional<SortOrder> sortOrder) {
+
+        this.id = id;
+        this.name = name;
+        this.parentId = parentId;
+        this.role = role;
+        this.sortOrder = sortOrder;
+    }
+
+    public Optional<String> getId() {
+        return id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public Optional<String> getParentId() {
+        return parentId;
+    }
+
+    public Optional<Role> getRole() {
+        return role;
+    }
+
+    public Optional<SortOrder> getSortOrder() {
+        return sortOrder;
+    }
+
+
+    @Override
+    public final boolean equals(Object obj) {
+        if (obj instanceof MailboxRequest) {
+            MailboxRequest other = (MailboxRequest) obj;
+            return Objects.equals(this.id, other.id)
+                && Objects.equals(this.name, other.name)
+                && Objects.equals(this.parentId, other.parentId)
+                && Objects.equals(this.role, other.role)
+                && Objects.equals(this.sortOrder, other.sortOrder);
+        }
+        return false;
+    }
+
+    @Override
+    public final int hashCode() {
+        return Objects.hash(id, name, parentId, role, sortOrder);
+    }
+
+    @Override
+    public String toString() {
+        return com.google.common.base.Objects.toStringHelper(getClass())
```

Comment:
If MoreObjects is available with gauva 16.0 please prefer using it....

It is deprecated using guava 18.0, and we might introduce warnings when merging MAILBOX-266